auth-providers/oauth2-provider/src/main/java/tech/ydb/auth/OAuth2TokenExchangeProvider.java

```diff
@@ -0,0 +1,320 @@
+package tech.ydb.auth;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.UnsupportedEncodingException;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
+import org.apache.http.HttpEntity;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tech.ydb.core.Status;
+import tech.ydb.core.StatusCode;
+import tech.ydb.core.UnexpectedResultException;
+import tech.ydb.core.auth.BackgroundIdentity;
+import tech.ydb.core.impl.auth.GrpcAuthRpc;
+
+/**
+ *
+ * @author Aleksandr Gorshenin
+ */
+public class OAuth2TokenExchangeProvider implements AuthRpcProvider<GrpcAuthRpc> {
+    public static final String GRANT_TYPE = "urn:ietf:params:oauth:grant-type:token-exchange";
+
+    private static final Logger logger = LoggerFactory.getLogger(OAuth2TokenExchangeProvider.class);
+    private static final Gson GSON = new Gson();
+
+    private final Clock clock;
+    private final String endpoint;
+    private final OAuth2TokenSource tokenSource;
+    private final List<NameValuePair> httpForm;
+    private final int timeoutSeconds;
+
```

Comment:
I can't see a scope and resource parameters in token exchange request:
https://github.com/ydb-platform/ydb-python-sdk/blob/main/ydb/oauth2_token_exchange/token_exchange.py#L66
https://github.com/ydb-platform/ydb-python-sdk/blob/main/ydb/oauth2_token_exchange/token_exchange.py#L80